tests/flytekit/unit/types/structured_dataset/test_structured_dataset.py

```diff
@@ -643,3 +660,43 @@ def wf_with_input() -> pd.DataFrame:
 
     pd.testing.assert_frame_equal(wf_no_input(), default_val)
     pd.testing.assert_frame_equal(wf_with_input(), input_val)
+
+
+
+def test_read_sd_from_uri(local_tmp_pqt_file):
+
+    @task
+    def upload_pqt_to_s3(local_path: str, remote_path: str) -> None:
+        """Upload local temp parquet file to s3 object storage"""
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            fs = FileAccessProvider(
+                local_sandbox_dir=tmp_dir,
+                raw_output_prefix="s3://my-s3-bucket"
+            )
+            fs.upload(local_path, remote_path)
+
+    @task
+    def read_sd_from_uri(uri: str) -> pd.DataFrame:
+        sd = StructuredDataset(uri=uri, file_format="parquet")
+        df = sd.open(pd.DataFrame).all()
+
+        return df
+
```

Comment:
Hi, @JiangJiaWei1103 
you should put these tasks in a workflow, 
which is closer to the reality that users use it.

Reply:
No problem, let me modify the test snippets.